packages/react/src/auto/AutoForm.ts

```diff
@@ -96,10 +97,24 @@ export const useFormFields = (
       )
     );
 
-    return [...includedObjectFields, ...includedRootLevelFields];
+    const allFormFields = [...includedObjectFields, ...includedRootLevelFields];
+    validateFormFieldApiIdentifierUniqueness(allFormFields.map(({ metadata }) => metadata.apiIdentifier));
+
+    return allFormFields;
   }, [metadata, options]);
 };
 
+const validateFormFieldApiIdentifierUniqueness = (apiIdentifiers: string[]) => {
+  const seen = new Set<string>();
+
+  for (const apiId of apiIdentifiers) {
+    if (seen.has(apiId)) {
+      throw new Error(`Input "${apiId}" appears multiple times in the action metadata. Input names must be unique.`);
```

Comment:
"action metadata" is more like internal stuff, and this error can be thrown when the user sets up their model action with duplicated input parameters, not when our side has some unexpected behaviours, so I think this error message is quite confusing for users.

How about a message like "Action iput parameter ${apiId} must be unique." or something similar?

Reply:
Updated to be 
```
`Input "${apiId}" is not unique for action "${actionApiIdentifier}"`
```